packages/typography/src/elements/CodeBlock.tsx

```diff
@@ -60,26 +68,52 @@ export const CodeBlock = React.forwardRef<HTMLPreElement, ICodeBlockProps>(
         <Highlight Prism={Prism} code={code ? code.trim() : ''} language={language || 'tsx'}>
           {({ className, tokens, getLineProps, getTokenProps }) => (
             <StyledCodeBlock className={className} ref={ref} isLight={isLight} {...other}>
-              {tokens.map((line, index) => (
-                <StyledCodeBlockLine
-                  {...getLineProps({ line })}
-                  key={index}
-                  isHighlighted={highlightLines && highlightLines.includes(index + 1)}
-                  isLight={isLight}
-                  isNumbered={isNumbered}
-                  size={_size}
-                >
-                  {line.map((token, tokenKey) => (
-                    <StyledCodeBlockToken
-                      {...getTokenProps({ token })}
-                      key={tokenKey}
-                      isLight={isLight}
-                    >
-                      {token.empty ? '\n' : token.content}
-                    </StyledCodeBlockToken>
-                  ))}
-                </StyledCodeBlockLine>
-              ))}
+              {tokens.map((line, index) => {
+                const getDiff = () => {
+                  let retVal: DIFF | undefined;
+
+                  if (language === 'diff') {
+                    const token = line.find(value => !(value.empty || value.content === ''));
+
+                    if (token) {
+                      if (token.types.includes('deleted')) {
+                        retVal = 'delete';
+                      } else if (token.types.includes('inserted')) {
+                        retVal = 'add';
+                      } else if (token.types.includes('coord')) {
+                        retVal = 'hunk';
+                      } else if (token.types.includes('diff')) {
+                        retVal = 'change';
+                      }
+                    }
+                  }
+
+                  return retVal;
+                };
```

Comment:
Could `getDiff` be moved into the component function scope and accept a `line` argument so that `getDiff` does not have to be defined for each `line` in the map function scope?

Reply:
Hmm, I had it that way, but didn't have access to prism-react-renderer's `Token[]` interface. Here the scoped interface is implicit. I'll take a closer look and see what I can come up with.

Reply:
I saw that the `Token` type wasn't exported. It may have to defined it in the file (which seemed to work). It's a small type, but maybe the juice isn't worth the squeeze. 

Just a thought, not a blocker, so I'll resolve this.